File: backend/src/main/java/com/graccasoft/schoolinvoicing/model/Invoice.java
package com.graccasoft.schoolinvoicing.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@Entity
@Getter
@Setter
public class Invoice extends BaseEntity  {
    private BigDecimal totalAmount;

    @ManyToOne
    @JoinColumn(name="student_id")
    private Student student;

    @ManyToOne
    @JoinColumn(name ="user_id")
    private User user;

    @OneToMany(mappedBy = "invoice", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<InvoiceItem> items;
}

File: frontend/src/app/model/invoice.ts
import { InvoiceItem } from "./Invoice-item"

export class Invoice{
    id!: number
    createdAt!: string
    studentId!: number
    studentName!: string
    schoolClassId!: number
    schoolClassName!: string
    totalAmount!: number
    items!: InvoiceItem

}